node/src/chain.rs

```diff
@@ -161,6 +191,11 @@ impl<N: Network, DB: database::DB, VM: vm::VMExecution>
                     let msg = recv?;
                     network.read().await.broadcast(&msg).await;
                 },
+                // Handles accept_block_timeout event
+                _ = sleep_until(timeout) => {
```

Comment:
If this select is biased, the timeout branch should go right after the result_chan receiver

Reply:
Why is that needed?

Reply:
You introduced `biased` and this means that the match branches are polled in sequential order.
In this case, you will process the first three branches even if the timeout is expired.

I was suggesting to move the timeout branch to the second place, so to give priority only to the "new block" messages.

However, if the "is_idle" function is supposed to be triggered when no messages are received at all, then it's ok to have the timeout as last branch

Reply:
Moving it to the second place, `after the result_chan receiver`, gives priority to a new block that originates from local consensus task while still discarding the new_blocks from the wire.

Reply:
Let's keep this then